[19.0][ADD] partner_identification_automation_activity

bosd opened the pull request:
Depends on:
#2218
#2224

bosd committed:
[ADD] partner_identification_automation_activity

bosd committed:
[DO NOT MERGE]

bosd committed:
Update test-requirements.txt to remove a dependency

Removed odoo-addon-partner_identification dependency.

BhaveshHeliconia commented on partner_identification_automation_activity/models/res_partner_identification.py:
```diff
@@ -0,0 +1,211 @@
+class ResPartnerIdNumber(models.Model):
+    _inherit = "res.partner.id_number"
+
+    def create(self, vals_list):
```
Missing `@api.model_create_multi`

BhaveshHeliconia commented on partner_identification_automation_activity/models/res_partner_identification.py:
```diff
@@ -0,0 +1,211 @@
+
+        if "status" in vals and vals["status"] == "pending":
+            # Get old status values before the write operation
+            self.env.cr.execute(
```
Avoid raw SQL query

BhaveshHeliconia commented on partner_identification_automation_activity/data/mail_activity_type.xml:
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
```
Add `noupdate="1"`

BhaveshHeliconia commented on partner_identification_automation_activity/models/res_partner_identification.py:
```diff
@@ -0,0 +1,211 @@
+        activities_vals = []
+
+        # Reload records to ensure related fields are properly loaded
+        records_to_process = self.browse(self.ids)
```
Redundant. Directly use `self`.

BhaveshHeliconia commented on partner_identification_automation_activity/models/res_partner_identification.py:
```diff
@@ -0,0 +1,211 @@
+            if existing_activity:
+                continue  # Skip if activity already exists
+
+            assigned_user = record.category_id.responsible_user_id or self.env.user
```
Use `category` variable

Commit:
[IMP] partner_identification_automation_activity: address review feedback

- Decorate create() with @api.model_create_multi.
- Replace the raw SQL old-status lookup in write() with an ORM read.
- Add noupdate="1" to data/mail_activity_type.xml.
- Drop redundant self.browse(self.ids)/browse(records.ids) reloads; iterate
  the recordsets directly.
- Reuse the already-resolved category variable in _create_renewal_activities.

bosd commented:
@BhaveshHeliconia thanks for the review! All addressed in a4c7f3644:

- **Missing `@api.model_create_multi`** — added the decorator on `create()`.
- **Avoid raw SQL** — replaced the `cr.execute` old-status lookup in `write()` with an ORM read (`{record.id: record.status for record in self}`).
- **Add `noupdate="1"`** — set on `data/mail_activity_type.xml`.
- **Redundant, use `self`** — dropped the `self.browse(self.ids)` reload (and the equivalent `self.browse(records.ids)` in `create()`); iterating the recordsets directly now.
- **Use `category` variable** — `_create_renewal_activities` now reuses the already-resolved `category` instead of `record.category_id`.

Could you re-review when you get a chance? 🙏

BhaveshHeliconia reviewed (approved):
LGTM!

Commit:
[IMP] partner_identification_automation_activity: cover activity-type fallbacks

Raise model coverage (~80% -> high) by exercising the previously untested
defensive paths:
- create()/_create_renewal_activities fallback to the generic 'To Do' type
  (by ref and by name search) and the no-type-available skip, forced via a
  patched env.ref so the module's own data records appear missing.
- renewal deadline when the category has no renewal lead time.
- renewal summary already containing the ID name.

Also drop the redundant inner 'if lead_unit and lead_number' (always true
under the outer condition).

Commit:
[FIX] partner_identification_automation_activity: drop fragile To-Do search tests

The two *_falls_back_to_todo_search tests assumed the generic activity type
is named exactly 'To Do', which isn't the case in 19.0, so the model's name
search returns nothing. The *_no_type_available tests already exercise the
search branch robustly (forcing an empty result), so remove the redundant
fragile ones.
